### ISSUE TITLE
Auth endpoint times out half the time

### ISSUE THREAD

**Issue by Tomino2112** [Microsoft/BotBuilder#2540]
Hi,

I have spent nearly whole day trying to figure out what is wrong with our bot. It works fine, our side of the implementation is ok, but every now and then (quite often) the bot just freezes. I have dived into your NodeJS library and after a lot of debugging it seems that the core issue is that the `authenticatedRequest` times out quite often. This is happening mainly when getting/posting user data, but it happened on other occasions as well. 

When it fails you can see this in console:
```
 Error: connect ETIMEDOUT 52.164.250.133:443
    at Object.exports._errnoException (util.js:1036:11)
    at exports._exceptionWithHostPort (util.js:1059:20)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1080:14)
```

Unfortunately it always fails at different times and places and I am unable to give clear instructions as how to reproduce it. It is worth noting that this became issue only after we moved our bot from localhost (ngrok) to production server (AWS).

This timeout issue is the main issue. Another issue that I have discovered is that sometime inner "throttling" kicks in and requests are just not triggered. This is caused by https://github.com/Microsoft/BotBuilder/blob/master/Node/core/lib/Session.js#L370 

There is no handler to try and repeat the request, bot just hangs.

I will keep debugging and see if I can give you some more info

**Comment by nwhitmont** [comment]
@Tomino2112 - thanks for reporting this issue.  Can you provide a code example that I can use to reproduce this timeout bug?

**Comment by nwhitmont** [comment]
@Tomino2112 are you still experiencing the issue using botbuilder v3.7.0?

**Comment by Tomino2112** [comment]
@nwhitmont Hi, sorry was not available lately. Cannot provide code as there is nothing specific in the code, pretty standard stuff. The code works fine on local over ngrok, once moved to our prod servers (AWS) we experienced this issue. 

I will test with `v3.7.0` this week and let you know

**Comment by nwhitmont** [comment]
@Tomino2112 reopen if you're still experiencing the issue
